### DISCUSSION TITLE
Failed Tasks automatic restart in Connectors

### DISCUSSION THREAD

**Discussion by AishD3** [strimzi/strimzi-kafka-operator#9756] [General]
I am writing with respect to connector and tasks restarts . Would like to know , what would be the difference between Solution in
https://strimzi.io/blog/2023/01/25/auto-restarting-connectors/ and Proposal document which is recently added here https://github.com/strimzi/proposals/blob/main/007-restarting-kafka-connect-connectors-and-tasks.md.

**Comment by scholzj** [comment]
What kind of difference do you have in mind? The auto-restarting of connectors as described in the blog post is a subset of the proposal you linked.

**Reply by AishD3** [reply]
Does the existing autorerstart: Enabled true will it automatically restart failed tasks alone.

**Reply by AishD3** [reply]
Example when we enable autoRestart to true 
Will it restart all tasks 
<img width="929" alt="image" src="https://github.com/strimzi/strimzi-kafka-operator/assets/142221898/3dc7b1e5-c161-4cd1-a099-5f7427f9e888">

or 
only failed tasks like this  /restart?includeTasks=true&onlyFailed=true

**Reply by scholzj** [reply]
It should restart only failed tasks as far as I know, yes.

**Reply by AishD3** [reply]
Is it possible to test once in your side , because when we used this feature and when we wanted to restart only failed tasks , This config was restarting all the tasks or the worker , because of which we have some impact in the connector that we use.

**Reply by scholzj** [reply]
I'm sorry, I do not have time to test it, you would have to test it yourself. The source code says it should restart only failed.
